architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,247 @@
+# Radius UCP Component Threat Model
+
+- **Author**: ytimocin
+
+## Overview
+
+This document provides a threat model for the Radius UCP (Universal Control Plane) component. It identifies potential security threats to this critical part of Radius and suggests possible mitigations. The document includes an analysis of the system, its assets, identified threats, and recommended security measures to protect the system.
+
+One of the most important goals of Radius is to be cloud agnostic and to enable portability of applications across platforms and environments. Therefore, the Radius UCP (Universal Control Plane) component helps the user apply ARM capabilities to non-Azure hosting models like AWS, GCP, and Kubernetes.
+
+In brief, UCP is a lightweight proxy that stands in front of the resource providers like Applications RP, AWS, and Azure. Based on the requests UCP gets, it forwards those requests to the appropriate service.
+
+## Terms and Definitions
+
+| Term                    | Definition                                                                                                                                  |
+| ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+| Radius Tracked Resource | Ex: `/planes/radius/local/resourceGroups/test-group/providers/System.Resources/resources/test-app-303153687ee5adbcf353bc6c2caa4373f31e04c6` |
```

Comment:
This isn't really sufficient to define what a tracked resource is. Is there a way we can talk about the security concerns without an extensive discussion of what the feature is?

Reply:
I agree. I will update the definition of the Tracked Resource. Not sure if that is what you mean, but we can also not talk about Tracked Resource, I guess.

Reply:
When I see a term in the definitions, I'd expect to see a clear definition. As a reader, I'd want to make sure I understand the term before reading the document.

I don't think we need every reader of the document to understand what a tracked resource is. I think the content in the implementation details section is enough, so you could remove this.